apm-agent-core/src/main/java/co/elastic/apm/agent/report/IntakeV2ReportingEventHandler.java

```diff
@@ -132,77 +167,43 @@ private void writeEvent(ReportingEvent event) {
         if (connection != null) {
-            if (reporter != null) {
-                timeoutTask = new FlushOnTimeoutTimerTask(reporter);
+            if (timeoutTask != null) {
                 if (logger.isDebugEnabled()) {
                     logger.debug("Scheduling request timeout in {}", reporterConfiguration.getApiRequestTime());
                 }
-                timeoutTimer.schedule(timeoutTask, reporterConfiguration.getApiRequestTime().getMillis());
+                timeoutTimer.schedule(timeoutTask, reporterConfiguration.getApiRequestTime().getMillis(), TimeUnit.MILLISECONDS);
             }
         }
         return connection;
     }
 
-    @Override
-    public void endRequest() {
-        cancelTimeout();
-        super.endRequest();
-    }
-
     @Override
     public void close() {
         super.close();
         logger.info("Reported events: {}", reported);
         logger.info("Dropped events: {}", dropped);
-        timeoutTimer.cancel();
+        timeoutTimer.shutdownNow();
     }
 
-    private static class FlushOnTimeoutTimerTask extends TimerTask {
-        private static final Logger logger = LoggerFactory.getLogger(FlushOnTimeoutTimerTask.class);
+    /**
+     * Schedules a wakeup event to the disruptor.
+     * This ensures that even in quiet periods, the event handler will be woken up to end the request after api_request_time has expired.
+     */
+    private static class WakeupOnTimeout implements Runnable {
         private final ApmServerReporter reporter;
-        @Nullable
-        private volatile Future<Void> flush;
 
-        private FlushOnTimeoutTimerTask(ApmServerReporter reporter) {
+        private WakeupOnTimeout(ApmServerReporter reporter) {
             this.reporter = reporter;
         }
 
         @Override
         public void run() {
-            logger.debug("Request flush because the request timeout occurred");
-            try {
-                // If the ring buffer is full this throws an exception.
-                // In case it's full due to a traffic spike it means that it will eventually flush anyway because of
-                // the max request size, but we need to catch this Exception otherwise the Timer thread dies.
-                flush = reporter.flush();
-            } catch (Exception e) {
-                // This shouldn't reoccur when the queue is full due to lack of communication with the APM server
-                // as the TimerTask wouldn't be scheduled unless connection succeeds.
-                logger.info("Failed to register a Flush event to the disruptor: {}", e.getMessage());
-            }
-        }
-
-        @Override
-        public boolean cancel() {
-            final boolean cancel = super.cancel();
-            final Future<Void> flush = this.flush;
-            if (flush != null) {
-                flush.cancel(false);
-            }
-            return cancel;
+            reporter.scheduleWakeupEvent();
```

Comment:
Still makes sense to wrap within try-catch, so that the thread doesn't die from any reason